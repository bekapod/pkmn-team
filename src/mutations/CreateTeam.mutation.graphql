mutation CreateTeam($name: String!) {
  createTeam(object: { name: $name }) {
    ...TeamFragment
  }
}
