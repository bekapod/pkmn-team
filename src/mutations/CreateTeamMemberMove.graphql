mutation CreateTeamMemberMove($memberId: uuid!, $moveId: uuid!, $order: Int!) {
  createTeamMemberMove(
    object: { team_member_id: $memberId, move_id: $moveId, order: $order }
  ) {
    ...TeamMemberMoveFragment
    team_member {
      id
      team {
        id
      }
    }
  }
}
