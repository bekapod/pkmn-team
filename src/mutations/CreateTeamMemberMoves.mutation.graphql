mutation CreateTeamMemberMoves($moves: [team_member_move_insert_input!]!) {
  createTeamMemberMoves(
    objects: $moves
    on_conflict: {
      constraint: team_member_move_team_member_id_move_id_key
      update_columns: [order]
    }
  ) {
    returning {
      ...TeamMemberMoveFragment
      team_member {
        id
        team {
          id
        }
      }
    }
  }
}
