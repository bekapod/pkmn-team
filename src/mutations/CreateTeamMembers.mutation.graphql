mutation CreateTeamMembers($members: [team_member_insert_input!]!) {
  createTeamMembers(objects: $members) {
    returning {
      ...TeamMemberFragment
    }
  }
}
