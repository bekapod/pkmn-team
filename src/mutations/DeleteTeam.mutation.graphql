mutation DeleteTeam($id: ID!) {
  deleteTeam(id: $id) {
    ...team
  }
}
