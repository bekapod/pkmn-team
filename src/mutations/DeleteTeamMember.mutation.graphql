mutation DeleteTeamMember($id: ID!) {
  deleteTeamMember(id: $id) {
    ...teamMember
  }
}
