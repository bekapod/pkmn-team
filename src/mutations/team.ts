import gql from "graphql-tag";

export const createTeam = gql`
  mutation($name: String!, $pokedexIds: [Int!]!) {
    createTeam(name: $name, pokedexIds: $pokedexIds) {
      id
    }
  }
`;
