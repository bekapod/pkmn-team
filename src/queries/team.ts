import gql from "graphql-tag";

export const getAllTeams = gql`
  query {
    allTeams {
      id
      name
      createdAt
      members {
        id
        pokemon {
          pokedexId
          name
          types
          sprite
        }
      }
    }
  }
`;
