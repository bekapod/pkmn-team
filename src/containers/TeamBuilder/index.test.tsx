import React from "react";
import { MockedProvider, MockedResponse } from "react-apollo/test-utils";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
// tslint:disable-next-line:no-implicit-dependencies
import { render } from "react-testing-library";
// tslint:disable-next-line:no-implicit-dependencies
import wait from "waait";
import TeamBuilderContainer from ".";
import { createTeam, updateTeam } from "../../mutations/team";
import { getTeamById } from "../../queries/team";
import configureStore from "../../store";

const mocks: ReadonlyArray<MockedResponse> = [
  {
    request: {
      query: createTeam,
      variables: {
        name: "Test Team",
        pokedexIds: [25]
      }
    },
    result: {
      createTeam: {
        id: "1"
      }
    }
  },

  {
    request: {
      query: updateTeam,
      variables: {
        id: "2",
        name: "Test Team 2",
        pokedexIds: [25, 25]
      }
    },
    result: {
      updateTeam: {
        id: "2"
      }
    }
  }
];

describe("<TeamBuilderContainer />", () => {
  it("renders team creation form", async () => {
    const { queryByPlaceholderText } = render(
      <Provider store={configureStore({})}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <MemoryRouter initialEntries={["/team/create/"]}>
            <TeamBuilderContainer />
          </MemoryRouter>
        </MockedProvider>
      </Provider>
    );

    await wait(0);

    expect(queryByPlaceholderText(/Find by name/)).toBeTruthy();
  });

  it("renders team edit form", async () => {
    const mocksWithTeam: ReadonlyArray<MockedResponse> = [
      ...mocks,
      {
        request: {
          query: getTeamById,
          variables: {
            id: "cji6gz8gwhblk0a9639btq2hd"
          }
        },

        result: {
          data: {
            teamById: {
              createdAt: "2018-06-08T21:15:14.723Z",
              id: "cji6gz8gwhblk0a9639btq2hd",
              members: [
                {
                  id: "cji6gz8gwhbll0a96aahx3ivv",
                  pokemon: {
                    name: "bulbasaur",
                    pokedexId: 1,
                    sprite: "1.png",
                    types: ["POISON", "GRASS"]
                  }
                },
                {
                  id: "cji6gz8gwhblm0a96eja18t10",
                  pokemon: {
                    name: "charmander",
                    pokedexId: 4,
                    sprite: "4.png",
                    types: ["FIRE"]
                  }
                },
                {
                  id: "cji6gz8gwhbln0a96q7wmx9zj",
                  pokemon: {
                    id: "7",
                    name: "squirtle",
                    pokedexId: 7,
                    sprite: "7.png",
                    types: ["WATER"]
                  }
                },
                {
                  id: "cji6gz8gwhblo0a96wgoki379",
                  pokemon: {
                    id: "25",
                    name: "pikachu",
                    pokedexId: 25,
                    sprite: "25.png",
                    types: ["ELECTRIC"]
                  }
                }
              ],
              name: "Starters Team"
            }
          }
        }
      }
    ];
    const team = mocksWithTeam[2].result.data.teamById;
    const { queryByDisplayValue, queryAllByTestId, queryByText } = render(
      <Provider store={configureStore({})}>
        <MockedProvider mocks={mocksWithTeam} addTypename={false}>
          <MemoryRouter initialEntries={[`/team/edit/${team.id}`]}>
            <TeamBuilderContainer match={{ params: { teamId: team.id } }} />
          </MemoryRouter>
        </MockedProvider>
      </Provider>
    );

    await wait(0);

    expect(queryByDisplayValue(team.name)).toBeTruthy();
    expect(queryAllByTestId(/pokemon-(\w+)/)).toHaveLength(team.members.length);
    expect(queryByText(/Save team/)).toBeTruthy();
  });
});
