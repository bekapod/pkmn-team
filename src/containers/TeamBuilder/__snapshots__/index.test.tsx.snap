// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamBuilderContainer /> renders team builder 1`] = `
Array [
  <h1
    className="Heading-b3adig-0 jwQDLL"
  >
    Create a Team
  </h1>,
  <div
    aria-live="assertive"
    className="SectionContainer-xjlkvu-0 jZbaLj"
    role="alert"
  >
    <div
      className="CenteredRow-f3rx2f-0 hfFVle"
    >
      <input
        aria-label="Choose a team name"
        className="TextInput-sc-149814h-0 GiantInput-sc-1mhmfj5-0 CEtIK"
        onChange={[Function]}
        placeholder="Choose a team name"
      />
    </div>
    <div
      className="CenteredRow-f3rx2f-0 hNpdbY"
    >
      <div>
        <div
          aria-expanded={false}
          aria-haspopup="listbox"
          aria-labelledby="downshift-0-label"
          aria-owns={null}
          className="Autocomplete-sc-1wak8g9-1 cILwaC"
          role="combobox"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-0-label"
            autoComplete="off"
            className="TextInput-sc-149814h-0 GiantInput-sc-1mhmfj5-0 CEtIK"
            id="downshift-0-input"
            onBlur={[Function]}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Choose a Pokemon"
            value=""
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
